### ISSUE TITLE
Clang reports -Wunused-member-function in unnamed namespace

### ISSUE THREAD

**Issue by AndrewPaxie** [rollbear/trompeloeil#57]
When defining a mock function in an unnamed namespace, for example,

```
namespace
{
  class self_ref_mock
  {
  public:
    void expect_self()
    {
      exp = NAMED_REQUIRE_CALL_0(*this, mfunc());
    }
    MAKE_MOCK0(mfunc, void());
    std::unique_ptr<trompeloeil::expectation> exp;
  };
}
``` 

Clang (all supported versions, all modes) reports the following warnings,

```
member function 'trompeloeil_self_mfunc' is not needed 
and will not be emitted [-Wunneeded-member-function]

member function 'trompeloeil_tag_mfunc' is not needed 
and will not be emitted [-Wunneeded-member-function]
```

Use these functions in preference to disabling this warning.

As a suggestion use them in the `MAKE_MOCKn` macros.
This is the preferred macro since they may never be referenced
in any other Trompeloeil context, such as when defining an 
expectation or using a modifier.

**Comment by rollbear** [comment]
Hrmm. At work now so I can't spend much time at the moment, but I'm pretty much prepared to file a bug report to clang for this. It *is* needed and it *is* used. That the only use is in a `decltype()` doesn't mean it not needed.

However, as you suggest, it's easy to work around.

**Comment by AndrewPaxie** [comment]
There's a pull request for your consideration now.

There are already two open problem reports regarding this warning, so it is already established that it is imperfect in its detection.  

BTW: "easy" is a relative term.  I tried many other workaround attempts - the closest I came to an "easy" solution was making the auxiliary functions `virtual`.  This makes me think their idea of "need" and "use" has something to do with the One Definition Rule.  I will look more closely at `decltype` to see whether is only a "potentially evaluated" context or somesuch refinement of our everyday notions of "use".
